docs/subscriptions/add-new.md

```diff
@@ -0,0 +1,289 @@
+      * If no `guid` is present, the server generates a `guid` for later use
+   2. Checks to see if there is an existing entry with the same `guid` or `feed_url`
+      * If an existing entry is found, the server sets the `is_subscribed` field to `true` and updates the `subscription_changed` date to the current date
+      * If no existing entry is found, the server creates a new subscription entry
+3. The server returns a success payload containing the subscription information for each object in the request payload.
+
+:::{mermaid}
+flowchart TB
+   post([The client posts a subscription payload]) --> process{Does the payload contain\n a GUID?}
+   subgraph each [For each object]
+      process -->|yes| store_guid(The server stores the GUID\nfor the feed)
+      process -->|no| generate_guid(The server generates a GUID\nfor the feed)
+      store_guid & generate_guid --> search_entries[[The server checks for existing\nentries with the same URL or GUID]]
+      search_entries --> found{Was an existing subscription entry found?}
+      found -->|yes| resubscribe(The server updates the is_subscribed field\nto true and updates the subscription_changed date\nto the current date)
+      found -->|no| create(The server creates a new subscription entry)
+   end
+   resubscribe & create --> return([The server returns a success payload])
+:::
+
+### Subscription GUID update
+
+If the client doesn't send a `guid` in the subscription payload, the server must create one immediately to ensure the following:
+
+1. Each entry has an associated `guid`
+2. The client receives a success response as quickly as possible
+
+Once this is done, the server should asynchronously verify that there isn't a more authoritative GUID available. The following flow should be used:
+
+1. The server checks for a more authoritative GUID using the following methods (in order)
+   1. Fetch and parse the RSS feed to search for a `guid` field
+   2. If the server admin has enabled Podcast Index support, send the RSS feed URL to the [`/podcast/byfeedurl` endpoint](https://podcastindex-org.github.io/docs-api/#get-/podcasts/byfeedurl) and check the response for a `podcastGuid` entry
+2. If a more authoritative `guid` is found, the server must update the subscription entry as follows:
+   1. Create a new subscription entry with the new `guid`
+   2. Update the `new_guid` field in the existing entry to point to the new `guid`
+   3. Update the `guid_changed` field in the existing entry to the current date
+
+:::{mermaid}
+flowchart TD
+   subgraph authority [For each empty GUID]
+      fetch(The server fetches the RSS feed and\nparses it) --> rss_guid{Does the RSS field contain a GUID?}
+      rss_guid -->|no| podcast_index{Has the server admin enabled Podcast Index\nsupport?}
+      podcast_index -->|yes| fetch_response(The server sends the RSS feed URL\nto the /podcast/byfeedurl endpoint)
+      fetch_response --> pi_guid{Does the response contain a `podcastGuid` entry?}
```

Comment:
@keunes It might be a good idea. From what I can see, if an RSS feed actually does have a GUID it will be very rare that it won't be in the feed itself. It would only happen if the feed itself were an outdated version. Podcast Index and other podcast hosts generate a missing GUID when they receive it, so we could probably cut this entirely.

Instead, if the server generates a GUID and the RSS feed receives one later, the client could check the feed periodically to see if this has changed and send a `PATCH` request to the server if it finds a more authoritative source.

> A podcast gets assigned a podcast:guid once in its lifetime using its current feed url (at the time of assignment) as the seed value. That GUID is then meant to follow the podcast from then on, for the duration of its life, even if the feed url changes. This means that when a podcast moves from one hosting platform to another, its podcast:guid should be discovered by the new host and imported into the new platform for inclusion into the feed.

Reply:
@keunes For now, I'd suggest we don't include this in the spec. This is something that a server dev could add as an optional feature, but client/RSS/server generated should be the core mechanisms used in all servers.